Connect4Project/app.js

```diff
@@ -0,0 +1,162 @@
+   // Creates the rest of the board, making new cells.
+   for (let y = 0; y < HEIGHT; y++) {
+     const row = document.createElement("tr");
+     for (let x = 0; x < WIDTH; x++) {
+       const cell = document.createElement("td");
+       cell.setAttribute("id", `${y}-${x}`);
+       row.append(cell);
+     }
+     board.append(row);
+   }
+ }
+ 
+ /** findSpotForCol: given column x, return top empty y (null if filled) */
+ 
+ function findSpotForCol(x) {
+   // TODO: write the real version of this, rather than always returning 0
+   // FOR REVIEW: I don't really understand the if statement, basically if a spot has not been filled, it returns the next empty slot in that column?
+   for(let y=HEIGHT - 1; y >= 0; y--){
+     if(!board[y][x]) {
+       return y;
+     }
+   }
+       return null;
+ }
+ 
+ /** placeInTable: update DOM to place piece into HTML table of board */
+ 
+ function placeInTable(y, x) {
+   // TODO: make a div and insert into correct table cell
+   // FOR REVIEW: In relation to the above review with the string template literal, i see how the pieces are added to the created cells here, but still unsure what the ID of the cells actually means. 
+   const piece = document.createElement("div")
+     piece.classList.add('piece');
+     piece.classList.add(`p${currPlayer}`);
+     piece.style.top = -50 * (y + 2);
+       const place = document.getElementById(`${y}-${x}`);
+         place.append(piece);
+   }
+ 
+ /** endGame: announce game end */
+ 
+ function endGame(msg) {
+   // TODO: pop up alert message
+   alert(msg)
+ }
+ 
+ /** handleClick: handle click of column top to play piece */
+ 
+ function handleClick(evt) {
+   // get x from ID of clicked cell
+ 
+   const x = +evt.target.id;
+ 
+   // get next spot in column (if none, ignore click)
+   const y = findSpotForCol(x);
+   if (y === null) {
+     return;
+   }
+ 
+   // place piece in board and add to HTML table
+   // TODO: add line to update in-memory board
+   board[y][x] = currPlayer
+   placeInTable(y, x);
+ 
+   // check for win
+   if (checkForWin()) {
+     return endGame(`Player ${currPlayer} won!`);
+   }
+ 
+   // check for tie
+   // TODO: check if all cells in board are filled; if so call, call endGame
+   if(board.every(row => row.every(cell => cell))) {
+     return endGame('Tie!');
+   }
+ 
+   // switch players
+   // TODO: switch currPlayer 1 <-> 2
+   currPlayer = currPlayer === 1 ? 2 : 1
+ }
+ 
+ /** checkForWin: check board cell-by-cell for "does a win start here?" */
+ 
+ function checkForWin() {
+   function _win(cells) {
+     // Check four cells to see if they're all color of current player
+     //  - cells: list of four (y, x) cells
+     //  - returns true if all are legal coordinates & all match currPlayer
+ 
+     return cells.every(
+       ([y, x]) =>
+         y >= 0 &&
+         y < HEIGHT &&
+         x >= 0 &&
+         x < WIDTH &&
+         board[y][x] === currPlayer
+     );
+   }
```

Comment:
This especially could use a comment, just explain what all of the conditions are.